File: backend/src/main/java/com/dollyanddot/kingmaker/domain/routine/dto/RoutineDto.java
package com.dollyanddot.kingmaker.domain.routine.dto;

import com.dollyanddot.kingmaker.domain.category.dto.CategoryDto;
import com.dollyanddot.kingmaker.domain.routine.domain.Routine;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class RoutineDto {

  private Long routineId;
  private Long categoryId;
  private String routineNm;
  private String routineDetail;
  private String period;
  private boolean importantYn;
  private LocalDateTime startAt;
  private LocalDateTime endAt;

  public static RoutineDto from(Routine routine){
    return RoutineDto.builder()
        .routineId(routine.getId())
        .categoryId(routine.getCategory().getId())
        .routineNm(routine.getName())
        .routineDetail(routine.getDetail())
        .period(routine.getPeriod())
        .importantYn(routine.isImportantYn())
        .startAt(routine.getStartAt())
        .endAt(routine.getEndAt())
        .build();
  }
}

File: frontend/kingmaker/lib/dto/routine_dto.dart
class RoutineDto{
  int routineId;
  int categoryId;
  String routineNm;
  String routineDetail;
  String period;
  bool importantYn;
  String startAt;
  String endAt;

  RoutineDto({
    required this.routineId,
    required this.categoryId,
    required this.routineNm,
    required this.routineDetail,
    required this.period,
    required this.importantYn,
    required this.startAt,
    required this.endAt,
  });

  factory RoutineDto.fromJson(Map<String, dynamic> json) {
    return RoutineDto(
      routineId: json['routineId'],
      categoryId: json['categoryId'],
      routineNm: json['routineNm'],
      routineDetail: json['routineDetail'],
      period: json['period'],
      importantYn: json['importantYn'],
      startAt: json['startAt'],
      endAt: json['endAt'],
    );
  }
  toRegistJson(int memberId) {
    return {
      "memberId" : memberId,
      "categoryId" : categoryId,
      "period" : period,
      "routineNm" : routineNm,
      "importantYn" : importantYn,
      "routineDetail" : routineDetail,
      "startAt" : startAt,
      "endAt" : endAt,
    };
  }

  String toString(){
    return " $routineId, $categoryId, $routineNm, $routineDetail, $period, $importantYn, $startAt, $endAt,";
  }

  toModifyJson() {
    return {
      "routineId" : routineId,
      "categoryId" : categoryId,
      "period" : period,
      "routineNm" : routineNm,
      "importantYn" : importantYn,
      "routineDetail" : routineDetail,
      "startAt" : startAt,
      "endAt" : endAt,
    };
  }
}